hommod_rest/services/interpro.py

```diff
@@ -96,99 +99,92 @@ def __init__(self, start, end, ac):
 
 _MAX_INTERPRO_JOBS = 30
 
+
 # Uses interproscan to obtain data: https://code.google.com/p/interproscan/wiki/HowToDownload
 class InterproService (object):
+    def __init__(self, storage_dir=None):
+        self._storage_dir = storage_dir
 
-    def __init__(self):
-
-        self.interproExe = None
-        self.storageDir = None
-        self._njobs = 0
-
-    def _checkinit (self):
-
-        from flask import current_app as flask_app
+    @property
+    def storage_dir(self):
+        return self._storage_dir
 
-#        if not self.interproExe:
-#            self.interproExe = flask_app.config ['INTERPROSCAN']
+    @storage_dir.setter
+    def storage_dir(self, storage_dir):
+        self._storage_dir = storage_dir
 
-        if not self.storageDir:
-            self.storageDir = flask_app.config ['INTERPRODIR']
+    def _checkinit(self):
+        if not self._storage_dir:
+            raise Exception("storage_dir not set")
 
-    def _interpro_lockfile_path (self, ID):
+    def _interpro_lockfile_path(self, ID):
 
-        return os.path.join (self.storageDir, 'lock_%s' % ID)
+        return os.path.join(self.storage_dir, 'lock_%s' % ID)
 
-    def _interpro_count_lockfiles (self):
+    def _interpro_count_lockfiles(self):
 
-        return len (glob (os.path.join (self.storageDir, "lock_*")))
+        return len(glob(os.path.join(self.storage_dir, "lock_*")))
 
     def _create_data_file(self, sequence):
 
-        self._checkinit ()
-        _log.info ("creating interpro file for sequence:\n%s" % sequence)
+        self._checkinit()
+        _log.info("creating interpro file for sequence:\n%s" % sequence)
 
-        if not self.storageDir:
-            raise Exception("storageDir must be set")
+        if not os.path.isdir(self.storage_dir):
+            os.mkdir(self.storage_dir)
 
-        if not os.path.isdir(self.storageDir):
-            os.mkdir(self.storageDir)
+        ID = idForSeq(sequence)  # unique
 
-        ID = idForSeq (sequence) # unique
```

Comment:
Rename to either `id` (lowercase) or `sequence_id` (more descriptive). If it's an accession code, use `ac_code`.